main.go

```diff
@@ -33,17 +40,31 @@ func main() {
 
 	// launch DNS server
 	if config.DnsOn {
-		resolver.LaunchDNS()
+		go func() {
+			if err := resolver.LaunchDNS(); err != nil {
+				logging.Error.Fatalf("DNS server failed: %v", err)
+			} else {
+				logging.Error.Fatalf("terminating because DNS died")
+			}
+		}()
 	}
 
 	// launch HTTP server
 	if config.HttpOn {
-		go resolver.LaunchHTTP()
+		go func() {
+			if err := resolver.LaunchHTTP(); err != nil {
+				logging.Error.Fatalf("HTTP server failed: %v", err)
```

Comment:
Wouldn't it be cleaner if the for launch{DNS, HTTP, ZK} happened within these functions?